scoped-transitions.md

```diff
@@ -0,0 +1,110 @@
+
+   Note: Pseudo-elements at this stage don't need to be developer exposed so UAs can use an alternate approach for this step. Using pseudo-elements (and default browser CSS) here helps define a consistent behavior for snapshots are updated if the scoped-transition-root is resized.
+
+3. The captured output above is painted in a stacking context which is a descendant of scoped-transition-root's stacking context. The above example generates the following stacking context hierarchy.
+
+   ```
+   root
+   └─ outer
+      ├─ inner
+      └─ outer-page-transition (pseudo-tree underneath)
+   ```
+
+   If the scoped-transition-root is tagged, it's snapshot uses [content-capture](https://github.com/WICG/shared-element-transitions/blob/main/explainer.md#more-granular-style-capture) mode. This means that only the element's descendants (including both text and elements) are painted in the snapshot while box decorations (like border, box-shadow) continue to paint in the outer stacking context.
+
+4. Once the `updateDOM` callback is finished, incoming-image pseudo-elements are added to the tree generated in step 2. Default animations are added as defined in [7.8](https://drafts.csswg.org/css-shared-element-transitions-1/#animate-a-page-transition-algorithm).
+
+## Constraints
+
+An element which is the root for a transition has the following constraints. These constraints may stop being satisfied at any phase of the transition lifecycle and we'll need to define the fallback behavior for such cases. The simplest option is to consider them a developer error and [skip the transition](https://drafts.csswg.org/css-shared-element-transitions-1/#skip-the-page-transition).
+
+1. There shouldn't be an active transition on the element, its descendants or its ancestors. This is because a transition involves taking the painted output of an element and rendering it into a pseudo-element. For example, in the following case:
+
+   ```html
+   <style>
+     .outer {
+       page-transition-tag: outer;
+     }
+     .inner {
+       page-transition-tag: inner;
+     }
+   </style>
+   <div class="outer">
+     <div class="inner"></div>
+   </div>
+   ```
+
+   ```js
+   document.createTransition(…);
+   document.querySelector('.outer').createTransition(…);
+   ```
+
+   There will be an image pseudo-element for `inner` in transitions on document and `outer` and its ambiguous which pseudo-element it should be painted in.
+
+2. The element which is the root for a transition must have `contain: layout`. This ensures it generates a stacking context and its painted output can be captured as an atomic unit.
+
+## Selector Syntax
```

Comment:
I think we can drop the section and save it for a shadow-dom-or-not comparison.

Reply:
Sounds good. I'll remove it. Is it worth keeping the point about why ::part is bad because of this feature though? The goal behind doing this design now was so we can point to it to veto the ::part syntax.